chanzuckerberg/corpora-data-portal
# Thuang 642 dropbox integration

fixes #642 #724 

Changes:

1. Adds `dropins.js` in `frontend/src/components/DropboxChooser/components/dropins.js` instead of using a script tag to load it, because it works better that way with React + code splitting

1. Extracts `EmptyDatasets` to its own Component file for organization

1. Creates `DropboxChooser` Component that accepts lazy loads the Chooser library when its children (button) are hovered over. It's also responsible for interacting with the Chooser library and edits/passes the dropbox link to its ancestor component via prop `onSelectUploadLink(dropboxLink)`

1. Creates POST upload-links mutation query

1. `views/Collection` Component now has `useEffect` that listens to new `dropboxLink` to make a POST call to upload the dataset

![demo](https://user-images.githubusercontent.com/6309723/101706631-6150ca00-3a3e-11eb-8006-6a98ab0d6338.gif)

<img width="726" alt="Screen Shot 2020-12-09 at 4 46 20 PM" src="https://user-images.githubusercontent.com/6309723/101706612-56963500-3a3e-11eb-8edd-573c358cce64.png">

QA Steps:

1. This is actually not easy to test locally, because Dropbox authentication doesn't like Requestly for some reason. So we have to deactivate Requestly, physically comment out some code to show the upload button on a Collection page even if the collection is not available (since all WRITE collections are private now, when we're logged out, the BE won't return the collection)

1. Open your dev console, so it will record the following network activities

1. Now you should be able to see the upload button and clicking on it will trigger the Dropbox window

1. Log in to your Dropbox account

1. Select an `.h5ad` file and click "Choose" button

1. Check network tab, and you should see a request to `***/upload-links` with a dropbox share link as `url`

OR: We can just merge this in `main` and test it on Dev lol

Thank you!

@Bento007 for read
@seve  for functionality
